certbot-nginx/certbot_nginx/obj.py

```diff
@@ -149,3 +150,36 @@ def __eq__(self, other):
                     self.path == other.path)
 
         return False
+
+    def has_redirect(self):
+        """Determine if this vhost has a redirecting statement
+        """
+        for directive_name in REDIRECT_DIRECTIVES:
+            found = _find_directive(self.raw, directive_name)
+            if found is not None:
+                return True
+        return False
+
+    def contains_list(self, test):
+        """Determine if raw server block contains test list at top level
+        """
+        for i in xrange(0, len(self.raw) - len(test)):
+            if self.raw[i:i + len(test)] == test:
+                return True
+        return False
+
+def _find_directive(directives, directive_name):
+    """Find a directive of type directive_name in directives
+    """
+    if len(directive_name) == 0 or not directives or \
+        isinstance(directives, str) or len(directives) == 0:
+        return None
+    else:
+        if directives[0] == directive_name:
+            return directives
+        else:
+            for line in directives:
```

Comment:
I don't understand how this would do the same thing. I want to DFS `directives`, using `None` as an indicator of failure to find a match in a subtree.